modules/gltf/SCsub

```diff
@@ -5,9 +5,12 @@ Import("env_modules")
 
 env_gltf = env_modules.Clone()
 
-# Godot source files
-env_gltf.add_source_files(env.modules_sources, "*.cpp")
-env_gltf.add_source_files(env.modules_sources, "structures/*.cpp")
+# Either add all the folders in one go if a SCU build...
+if env_gltf.add_source_files_scu(env.modules_sources, "*.cpp") == False:
+    # or if not a scu build, the above line only adds the main folder,
+    # and the subfolders are added separately:
```

Comment:
I'm confused, does `add_source_files(..., "*.cpp")` and `add_source_files_scu(..., "*.cpp")` behave differently to expand the wildcard?

If so that's not a very good design IMO, as it's error prone and confusing. `"*.cpp"` should normally expand only to files in the current folder, not recursively through subfolders.

It's also bug prone - if that's what you're doing here, you're adding the `editor/*.cpp` files twice, and compiling them in non-editor builds too.

Reply:
Sorry i didn't make clear above:

In this PR, in all cases _except_:

* modules/gltf
* modules/gltf/extensions

There is no recursion involved to subfolders, it only compiles the files in the folder. The system has _the ability_ to go into further folders, which is currently only used in this PR for these two cases. If desired, this feature can be removed. It will reduce the gains to compile time (particularly in later modules PRs), but that trade off may be worth it if it makes the system easier to understand.

`add_source_files_scu()` is actually a "dumb" command. It essentially just goes into the relevant `scu` folder and compiles all the files it finds. The actual decision as to which files to be compiled is made in the `scu_builders.py`, which maps out all this stuff. The `add_source_files_scu()` command is to maintain compatibility with the non-scu build.

In essence the problem this mechanism solves is that, particularly in third party, folders are used for organisation, but a small number of files in a folder is inefficient. We don't want to be compiling 1 or 2 files at a time, we want to be compiling 30 or 40 files at a time.

Reply:
I have the feeling that `add_source_files_scu()` might not be needed at all then, if everything can be controlled from `scu_builders.py`. You could just leave the normal `add_source_files()` call, but have the scu build enable a flag in that method that bypasses adding the sources at all, since `scu_builders.py` already handles it.

Like `scu_builders.py` registers some global with the files it already handled, and `add_source_files()` checks this and skips the matching files.

Reply:
Yup it's possible, I did try and evaluate this when I first wrote the system, I'll have another go. If `env["use_scu"]` is set, it could check a hash table or something like that.